backend/src/utils/logger.ts

```diff
@@ -0,0 +1,35 @@
+import { createLogger, format, transports } from 'winston';
+const { combine, timestamp, printf, errors } = format;
+
+// Define a custom log format
+const logFormat = printf(({ level, message, timestamp, stack }) => {
+  return `${timestamp} [${level.toUpperCase()}]: ${stack || message}`;
+});
+
+const logger = createLogger({
+  level: 'info', // Set default log level (info, warn, error)
+  format: combine(
+    timestamp(), // Add timestamps to log entries
+    errors({ stack: true }), // Log stack traces
+    logFormat, // Apply custom log format
+  ),
+  transports: [
+    // Log to a file
+    new transports.File({ filename: 'logs/error.log', level: 'error' }),
+    new transports.File({ filename: 'logs/combined.log' }),
+  ],
+});
+
+// Add console logging in development environment
+if (process.env.NODE_ENV !== 'production') {
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Refine environment check for console logging</b></div>
<div id="fix">

The condition `process.env.NODE_ENV !== 'production'` might not cover all non-production environments. Consider using a more explicit check for development environments.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -24 +24 @@
 -if (process.env.NODE_ENV !== 'production') {
 +if (process.env.NODE_ENV === 'development') {
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/Food-Ordering-App/pull/47#issuecomment-2439370000>#eef1bf</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged